torchvision/datasets/hpatches.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import print_function
+import resource
+import torch.utils.data as data
+from PIL import Image
+import os
+import errno
+import numpy as np
+import torch
+import tarfile
+import json
+import random
+import itertools
+from multiprocessing import Pool, Process, Manager
+from .utils import download_url, check_integrity
+from ..transforms import functional as F
+resource.setrlimit(resource.RLIMIT_NOFILE, (4096, 4096))
+
+splits_seqs = {'a': 'train', 'b': 'train', 'c': 'train', 'illum': 'test',
+               'view': 'test', 'full': 'test'}
+tps = ['ref', 'e1', 'e2', 'e3', 'e4', 'e5', 'h1', 'h2', 'h3', 'h4', 'h5',
+       't1', 't2', 't3', 't4', 't5']
+
+
+class HPatches(data.Dataset):
+    """`HPatches <https://hpatches.github.io/>`_ Dataset.
+    Args:
+        root (string): Root directory of dataset where ``hpatches-release``
+            will be downloaded.
+        split (string, optional): If given, only sequences from a specific split will be
+            used. Default is the 'full' split, i.e. all sequences.
+        transform (callable, optional): A function/transform that takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``.
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If the dataset is already downloaded, it is not
+            downloaded again.
+        output (string, optional): Type of output from the dataset generator: {pairs,
+            triplets, sets}. Pairs are suitable for siamese networks, triplets for
+            triplet networks, and sets for generic metric learning method. Default is
+            'pairs'.
+        n_samples (int, optional): Number of items to generate at each epoch. Default
+            is 1e7.
+        cache_to_ram (bool, optional): Whether to cache to all sequences to RAM to avoid
```

Comment:
Not, sure I am guessing 10gigs of ram are not that bad. 
Perhaps this can be dynamically set? I.e if there is more than 12 or so free ram cache, otherwise not

Reply:
OK, added automatically setting this to `False` if there is not enough available ram.

Reply:
Changed